host/sgx/create.c

```diff
@@ -832,6 +832,11 @@ oe_result_t oe_sgx_build_enclave(
         &loaded_enclave_pages_size,
         &enclave_size));
 
+    if (props.config.attributes & OE_SGX_FLAGS_KSS)
+    {
+        context->attributes.flags |= OE_SGX_FLAGS_KSS;
```

Comment:
Today the only possible bitmask flags for `context->attributes.flags` are `OE_ENCLAVE_FLAG_DEBUG (1)` and `OE_ENCLAVE_FLAG_SIMULATE (2)`. XOR it with OE_SGX_FLAGS_KSS (0x80) has no conflict today but it may in the future. I'd suggest we define another KSS flag value to go with `OE_ENCLAVE_FLAG_DEBUG (1)` and `OE_ENCLAVE_FLAG_SIMULATE (2)` but don't publish it.

Reply:
Another good catch @yentsanglee! I think this bug is not part of this code, but an existing one in [`oe_create_enclave`](https://github.com/openenclave/openenclave/blob/4ecb952ebca19932d28892ac2c84ba18a48574a2/host/sgx/create.c#L1018), where it's calling `oe_sgx_initialize_load_context` directly with flags using the `OE_ENCLAVE_FLAG_*` definitions, but the function (and the `context.attributes` member) expects flags defined using `OE_SGX_FLAGS_*`.

It should be the responsibility of the `oe_create_enclave` method to interpret the `OE_ENCLAVE_FLAG_*` into the appropriate `OE_SGX_FLAGS_DEBUG` (and whatever mechanism it will use to set the `OE_SGX_FLAGS_KSS`) before calling `oe_sgx_initialize_load_context` and the downstream methods should only have to deal with the `OE_SGX_FLAGS_*`.

Reply:
Added OE_SGX_FLAGS_KSS flag

Reply:
Possible bits for `context->attributes.flags` are:

```
#define OE_ENCLAVE_FLAG_DEBUG 0x00000001u
#define OE_ENCLAVE_FLAG_SIMULATE 0x00000002u
#define OE_ENCLAVE_FLAG_KSS 0x00000004u
```

Line 837 should set the bit of newly defined flag `OE_ENCLAVE_FLAG_KSS`. Without this change, the following statement will never be satisfied.

>> •	If SIGSTRUCT contains non-zero ISVFAMILYID/ISVEXTPRODID, loader must set SECS.ATTRIBUTES.KSS as 1

Reply:
Updated